include/hip/hcc_detail/hip_runtime_api.h

```diff
@@ -3055,6 +3055,65 @@ hipError_t hipSetupArgument(const void* arg, size_t size, size_t offset);
 hipError_t hipLaunchByPtr(const void* func);
 
 
+/**
+ * @brief Push configuration of a kernel launch.
+ *
+ * @param [in] gridDim   grid dimension specified as multiple of blockDim.
+ * @param [in] blockDim  block dimensions specified in work-items
+ * @param [in] sharedMem Amount of dynamic shared memory to allocate for this kernel.  The
+ * kernel can access this with HIP_DYNAMIC_SHARED.
+ * @param [in] stream    Stream where the kernel should be dispatched.  May be 0, in which case the
+ * default stream is used with associated synchronization rules.
+ *
+ * @returns hipSuccess, hipInvalidDevice, hipErrorNotInitialized, hipErrorInvalidValue
+ *
+ */
+
+hipError_t __hipPushCallConfiguration(dim3 gridDim,
+                                      dim3 blockDim,
+                                      size_t sharedMem __dparm(0),
+                                      hipStream_t stream __dparm(0));
+
+/**
+ * @brief Pop configuration of a kernel launch.
+ *
+ * @param [out] gridDim   grid dimension specified as multiple of blockDim.
+ * @param [out] blockDim  block dimensions specified in work-items
+ * @param [out] sharedMem Amount of dynamic shared memory to allocate for this kernel.  The
+ * kernel can access this with HIP_DYNAMIC_SHARED.
+ * @param [out] stream    Stream where the kernel should be dispatched.  May be 0, in which case the
+ * default stream is used with associated synchronization rules.
+ *
+ * @returns hipSuccess, hipInvalidDevice, hipErrorNotInitialized, hipErrorInvalidValue
+ *
+ */
+hipError_t __hipPopCallConfiguration(dim3 *gridDim,
```

Comment:
Curious to know why API name is starting with '_' as I could not see any API name has been started as '_hip'. I feel if it is always good to follow one convention.

Reply:
this corresponds to the undocumented cuda API `__cudaPushCallConfiguration`. it is used by compiler to implement the new kernel launching.

Reply:
It looks like `__cudaPushCallConfiguration` have to be hipified as well.

Reply:
usually __cudaPushCallConfiguration, __cudaPopCallConfiguration and cudaLaunchKernel  are only called in compiler generated binary, not in users' code. However, it does not hurt to hipify them. cudaLaunchKernel  corresponds to hipLaunchKernelByPtr.

Reply:
Ok, will add to hipify tools after merging of this PR.

Reply:
hipLaunchKernel was defined as a macro for user to launch kernel. Later it was replaced by hipLaunchKernelGGL. However some old HIP program may still contain call of hipLaunchKernel. This hipLaunchKernel only works as a macro. It has different semantics as the counterpart of cudaLaunchKernel. Since the name has already been taken, we have to use a different function name for the counterpart of cudaLaunchKernel. Since we used hipLaunchByPtr for cudaLaunch before, this time we use hipLaunchKernelByPtr for cudaLaunchKernel.

Reply:
> 
> 
> hipLaunchKernel was defined as a macro for user to launch kernel. Later it was replaced by hipLaunchKernelGGL. However some old HIP program may still contain call of hipLaunchKernel. This hipLaunchKernel only works as a macro. It has different semantics as the counterpart of cudaLaunchKernel. Since the name has already been taken, we have to use a different function name for the counterpart of cudaLaunchKernel. Since we used hipLaunchByPtr for cudaLaunch before, this time we use hipLaunchKernelByPtr for cudaLaunchKernel.

This has changed. We have reclaimed hipLaunchKernel , a C API for kernel launch- and it is already available in latest github master.

Reply:
OK I am going to use hipLaunchKernel. Currently it is in a conditional macro disabled for hip-clang, I will move it out of that condition.

Reply:
> 
> 
> OK I am going to use hipLaunchKernel. Currently it is in a conditional macro disabled for hip-clang, I will move it out of that condition.

Thanks !